### PR TITLE
docs: Add issue template

### PR DESCRIPTION
**Pull request by tanmay-pnaik**
## Overview

Add issue template

## Notes for reviewer

Adapted from https://github.com/bank-vaults/vault-operator/tree/main/.github/ISSUE_TEMPLATE

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 1]
```diff
@@ -0,0 +1,103 @@
+name: 🐛 Bug report
+description: Report a bug to help us improve vault-env
+labels: [bug]
```
Comment (sagikazarmark) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:3]:
```suggestion
labels: [kind/bug]
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        If you are not sure if it is a bug or not, you can contact us via the available [support channels](https://github.com/bank-vaults/vault-env/issues/new/choose).
+  - type: checkboxes
+    attributes:
+      label: Preflight Checklist
+      description: Please ensure you've completed all of the following.
+      options:
+        - label: I have searched the [issue tracker](https://www.github.com/bank-vaults/vault-env/issues) for an issue that matches the one I want to file, without success.
+          required: true
+        - label: I am not looking for support or already pursued the available [support channels](https://github.com/bank-vaults/vault-env/issues/new/choose) without success.
+          required: true
```
Comment (sagikazarmark) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:21]:
```suggestion
          required: true
        - label: I agree to follow the [Code of Conduct](https://bank-vaults.dev/docs/code-of-conduct/).
          required: true
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      placeholder: 1.20.0
+    validations:
+      required: true
+  - type: input
+    attributes:
+      label: Operator Version
+      description: What version of the Operator are you using?
+      placeholder: 1.20.0
+    validations:
+      required: true
```
Comment (sagikazarmark) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:35]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  - type: dropdown
+    attributes:
+      label: Installation Type
+      description: How did you install the Operator?
+      options:
+        - Official Helm chart
+        - Custom Helm chart
+        - Other (specify below)
```
Comment (sagikazarmark) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:43]: Maybe mention the webhook

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  - type: input
+    attributes:
+      label: Bank-Vaults Version
+      description: What version of the Bank-Vaults CLI are you using?
+      placeholder: leave empty if you haven't specified a custom version
```
Comment (sagikazarmark) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:48]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    attributes:
+      label: Configuration
+      description: Contents of Vault CustomResource.
+      render: yaml
+      placeholder: |
+        apiVersion: "vault.banzaicloud.com/v1alpha1"
+        kind: "Vault"
+        metadata:
+          name: "vault"
+        # ...
```
Comment (sagikazarmark) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:94]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+  - type: textarea
+    attributes:
+      label: Logs
+      description: Operator application logs (if relevant).
```
Comment (sagikazarmark) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:98]: vault-env?

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+        Please describe what you would like to change/add and why in detail by filling out the template below.
+
+        If you are not sure if your request fits into Dex, you can contact us via the available [support channels](https://github.com/bank-vaults/vault-env/issues/new/choose).
+  - type: checkboxes
+    attributes:
+      label: Preflight Checklist
+      description: Please ensure you've completed all of the following.
+      options:
+        - label: I have searched the [issue tracker](https://www.github.com/bank-vaults/vault-env/issues) for an issue that matches the one I want to file, without success.
+          required: true
```
Comment (sagikazarmark) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:18]:
```suggestion
          required: true
        - label: I agree to follow the [Code of Conduct](https://bank-vaults.dev/docs/code-of-conduct/).
          required: true
```

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 9]
```diff
@@ -0,0 +1,38 @@
+name: 🎉 Feature request
+description: Suggest an idea for the vault-env
```
Comment (sagikazarmark) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:2]:
```suggestion
description: Suggest an idea for the vault-env
labels: [kind/enhancement]
```

**Review by sagikazarmark — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 7 above)
Comment (tanmay-pnaik) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:77]: `vault-env` runs as an init container, right? Logs are deleted.

**Comment by tanmay-pnaik** [comment]
@ramizpolic Appreciate your eyes here.

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+labels: [kind/bug]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thank you for submitting a bug report!
+
+        Please fill out the template below to make it easier to debug your problem.
+
+        If you are not sure if it is a bug or not, you can contact us via the available [support channels](https://github.com/bank-vaults/vault-env/issues/new/choose).
```
Comment (ramizpolic) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:12]:
```suggestion
        If you are not sure if it is a bug or not, you can contact us via the available [support channels](https://bank-vaults.dev/docs/support/).
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+      label: Kubernetes Version
+      description: What version of Kubernetes are you using?
+      placeholder: 1.27.0
+    validations:
+      required: true
+  - type: input
+    attributes:
+      label: Kubernetes Distribution/Provisioner
+      description: Which Kubernetes distribution/privisioner are you using?
+      placeholder: e.g. GKE, EKS, AKS etc
```
Comment (ramizpolic) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:51]:
```suggestion
      placeholder: e.g. GKE, EKS, AKS, etc
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+      description: Steps to reproduce the behavior if it is not self-explanatory.
+      placeholder: |
+        1. In this environment...
+        2. With this config...
+        3. Run '...'
+        4. See error...
+  - type: textarea
+    attributes:
+      label: Logs
+      description: Operator application logs (if relevant).
```
Comment (ramizpolic) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:78]:
```suggestion
      description: Operator or application logs (if relevant).
```

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+labels: [kind/enhancement]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thank you for submitting a feature request!
+
+        Please describe what you would like to change/add and why in detail by filling out the template below.
+
+        If you are not sure if your request fits into Dex, you can contact us via the available [support channels](https://github.com/bank-vaults/vault-env/issues/new/choose).
```
Comment (ramizpolic) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:12]:
```suggestion
        If you are not sure if your request fits into Dex, you can contact us via the available [support channels](https://bank-vaults.dev/docs/support/).
```

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Alternatives Considered
+      description: A clear and concise description of any alternative solutions or features you've considered.
+  - type: textarea
+    attributes:
+      label: Additional Information
+      description: Add any other context about the problem here.
```
Comment (ramizpolic) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:41]:
```suggestion
      description: Add any other relevant context here.
```

**Review by ramizpolic — APPROVED** [review]

**Pull request merged**